packages/core/src/RPA/core/locators.py

```diff
@@ -34,6 +35,31 @@ def open_stream(obj, *args, **kwargs):
             obj.close()
 
 
+def find_by_name_or_error(path, criteria):
+    db = LocatorsDatabase(path)
+    if not Path(db.path).exists():
+        db.logger.warning("File does not exist: %s", db.path)
+
+    db.load()
+    if db.error:
+        error_msg, error_args = db.error
+        raise ValueError(error_msg % error_args)
```

Comment:
I think this should be ValidationError

Reply:
I was thinking of this ValidationError as something that should come up when a locator is invalid in format / data. They won't pop up here because db.load() resets validation errors:

```
 def load(self):
        """Deserialize database from file."""
        try:
            with open_stream(self.path, "r") as fd:
                data = json.load(fd)

            self._validate_all(data)
            self._locators = data
            self.reset_error()
```

Reply:
It resets if nothing before it raised an exception, but sets an error in the except branch

Reply:
Oh right of course. I'll change that to ValidaitonError then

Reply:
Changed at https://github.com/robocorp/rpaframework/pull/58/files